server/src/main/scala/za/co/absa/atum/server/api/service/PartitioningServiceImpl.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.service
+
+import za.co.absa.atum.model.dto.{AdditionalDataSubmitDTO, PartitioningSubmitDTO}
+import za.co.absa.atum.server.api.exception.ServiceError
+import za.co.absa.atum.server.api.repository.PartitioningRepository
+import za.co.absa.fadb.exceptions.StatusException
+import zio._
+
+class PartitioningServiceImpl(partitioningRepository: PartitioningRepository)
+  extends PartitioningService with BaseService {
+
+  override def createPartitioningIfNotExists(
+    partitioning: PartitioningSubmitDTO
+  ): IO[ServiceError, Either[StatusException, Unit]] = {
+    repositoryCallWithStatus(
+      partitioningRepository.createPartitioningIfNotExists(partitioning), "createPartitioningIfNotExists"
+    )
+  }
+
+  override def createOrUpdateAdditionalData(
+    additionalData: AdditionalDataSubmitDTO
+  ): IO[ServiceError, Either[StatusException, Unit]] = {
+    repositoryCallWithStatus(
+      partitioningRepository.createOrUpdateAdditionalData(additionalData), "createOrUpdateAdditionalData"
+    )
+  }
+
+}
+
+object PartitioningServiceImpl {
+  val layer: URLayer[PartitioningRepository, PartitioningService] = ZLayer {
+    for {
+      partitioningRepository <- ZIO.service[PartitioningRepository]
+    } yield new PartitioningServiceImpl(partitioningRepository)
+  }
+}
```

Comment:
Does this qualify for what we talked about for separating the classes, case class, traits and objects, or it is easy enough to   have the class and the object in one file like this?

Reply:
I actually separated it recently, but definitely good question / observation.

Before this module contained abstract class `PartitioningService`, and also concrete class `PartitioningServiceImpl`. Now it's split into 2 files, and the filenames correspond to the class/object names.

It's true that there is a class and object living in the same file, but that is pretty much a 'standard' or a common practice - because this object is a `companion object` to the class above, so it actually is greatly encouraged to live together. Companion objects are 'supplying' certain static methods / constants etc to a class that it belongs to. 

You can read more here: https://docs.scala-lang.org/overviews/scala-book/companion-objects.html